openedx/frontend-app-learning
# feat: option to show/hide ungraded assignment in progress page

Adds option to show ungraded assignment/problems in progress page.

* `Private-ref`: [BB-8824](https://tasks.opencraft.com/browse/BB-8824)
* Product roadmap ticket: https://github.com/openedx/platform-roadmap/issues/359
* Documentation PR: https://github.com/openedx/edx-documentation/pull/2268

**Test instructions:**

* Setup tutor nightly locally.
* Mount this repository using `tutor mounts add /path/to/frontend-app-learning`.
* Run `npm install` in this directory.
* Run `tutor dev launch -I`.
* Add `"SHOW_UNGRADED_ASSIGNMENT_PROGRESS": True,` to `MFE_CONFIG` dict `$(tutor config printroot)/env/apps/openedx/settings/lms/development.py`
* Go to progress tab in any course in learning mfe and confirm that ungraded assignments are displayed.

**Screenshots:**

`SHOW_UNGRADED_ASSIGNMENT_PROGRESS = True`

![image](https://github.com/openedx/frontend-app-learning/assets/10894099/47944130-6b28-416e-85a9-69e292cc2197)

![image](https://github.com/openedx/frontend-app-learning/assets/10894099/4297a217-62fb-4c7c-a32e-058b92a027a3)


`SHOW_UNGRADED_ASSIGNMENT_PROGRESS = False` (default)

![image](https://github.com/openedx/frontend-app-learning/assets/10894099/6cb786fa-5065-465e-98f5-33afbf92baab)
